modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -1769,7 +1770,138 @@ def sort_by(
         PandasDataframe
             A new PandasDataframe sorted into lexicographical order by the specified column(s).
         """
-        pass
+        columns = columns if isinstance(columns, list) else [columns]
+        axis = Axis(axis)
+        # The first if selects cases where we are either sorting by a single column that is not
+        # the index, or we are sorting by multiple columns, since in this case, we will have to
+        # shuffle the data before sorting.
+        # The elif selects the case where we are only sorting by index. All partitions maintain
+        # a copy of the index for their data, so if we are just sorting by index, we can make
+        # column partitions, and sort each by their index in parallel, which should be faster than
+        # shuffling our data and sorting.
+        if axis == Axis.ROW_WISE and not (
+            len(columns) == 1 and columns[0] == self.index.names[0]
+        ):
+
+            def sample_func(df):
+                """
+                Sample the given partition.
+
+                This function is applied to each row-axis partition in parallel to select samples
+                from each. It samples using the following algorithm:
+                    * If there are <= 100 rows in the dataframe, we pick quantiles over the entire dataframe.
+                    * If there are 100 < # of rows <= 1900, we pick quantiles over the first 100 rows, plus a sample
```

Comment:
Maybe I missed this in the design doc, but why are we still picking quantiles over the first 100 rows in this case?

Reply:
I decided to still pick quantiles since they would probably be a good sample, and save us from having to send over more data to the head node.

Reply:
This is a good point - if we end up using our original sampling algorithm, we'll need to discuss whether or not we should do this (although if we use the terasort algorithm, it shouldn't be a problem)